src/app/services/api.ts

```diff
@@ -0,0 +1,51 @@
+import axios from "axios";
+
+const API_URL = "http://localhost:8000/api";
+
+export const fetchMataKuliah = async () => {
+  try {
+    const response = await axios.get(`${API_URL}/course`);
+    return response.data;
+  } catch (error) {
+    console.error("Error fetching courses:", error);
+    throw error;
+  }
+};
+
+export const createMataKuliah = async (data: {
+  kode: string;
+  nama: string;
+  semester: string;
+  sks: string;
+}) => {
```

Comment:
Boleh ubah de pe "signature" dari function supaya nanti lebih gampang kalau mau tambah parameter

```suggestion
type CreateMataKuliahProps = {
  kode: string;
  nama: string;
  semester: string;
  sks: string;
}

export const createMataKuliah = async (data: CreateMataKuliahProps ) => {
```

Reply:
Btw, ini boleh pakai "commit suggesstion" kalau mau langsung di apply ke kodenya